src/pages/App.tsx

```diff
@@ -255,3 +207,59 @@ export default function App() {
     </ErrorBoundary>
   )
 }
+
+function UserPropertyUpdater() {
+  const isDarkMode = useIsDarkMode()
+
+  const [routerPreference] = useRouterPreference()
+  const userOptedOutOfUniswapX = useUserOptedOutOfUniswapX()
+  const isUniswapXDefaultEnabled = useUniswapXDefaultEnabled()
+
+  useEffect(() => {
+    // User properties *must* be set before sending corresponding event properties,
+    // so that the event contains the correct and up-to-date user properties.
+    user.set(CustomUserProperties.USER_AGENT, navigator.userAgent)
```

Comment:
are we sure these are also ok to move into the statsig provider? 👀 might run into the issue of these use effects running at the same time as other ones that send events I think? nbd would just keep an eye on it